pinot-common/src/main/java/org/apache/pinot/common/config/TagNameUtils.java

```diff
@@ -27,78 +28,93 @@ private TagNameUtils() {
-  }
-
-  private static String buildOfflineTagFromTenantName(String tenantName) {
-    return tenantName + "_" + ServerType.OFFLINE.toString();
-  }
-
-  private static String buildBrokerTenantTagFromTenantName(String tenantName) {
-    return tenantName + "_" + TenantRole.BROKER.toString();
+  public static boolean isBrokerTag(String tagName) {
+    return tagName.endsWith(BROKER_TAG_SUFFIX);
   }
 
   public static boolean isServerTag(String tagName) {
     return isOfflineServerTag(tagName) || isRealtimeServerTag(tagName);
   }
 
   public static boolean isOfflineServerTag(String tagName) {
-    return tagName.endsWith(ServerType.OFFLINE.toString());
+    return tagName.endsWith(OFFLINE_SERVER_TAG_SUFFIX);
   }
 
   public static boolean isRealtimeServerTag(String tagName) {
-    return tagName.endsWith(ServerType.REALTIME.toString());
+    return tagName.endsWith(REALTIME_SERVER_TAG_SUFFIX);
   }
 
-  public static boolean isBrokerTag(String tagName) {
-    return tagName.endsWith(TenantRole.BROKER.toString());
+  public static String getTenantFromTag(String tagName) {
+    if (tagName.endsWith(BROKER_TAG_SUFFIX)) {
+      return tagName.substring(0, tagName.length() - BROKER_TAG_SUFFIX.length());
+    }
+    if (tagName.endsWith(OFFLINE_SERVER_TAG_SUFFIX)) {
+      return tagName.substring(0, tagName.length() - OFFLINE_SERVER_TAG_SUFFIX.length());
+    }
+    if (tagName.endsWith(REALTIME_SERVER_TAG_SUFFIX)) {
+      return tagName.substring(0, tagName.length() - REALTIME_SERVER_TAG_SUFFIX.length());
+    }
+    return tagName;
+  }
+
+  public static String getBrokerTagForTenant(@Nullable String tenantName) {
+    return getTagForTenant(BROKER_TAG_SUFFIX, tenantName);
+  }
+
+  public static String getOfflineTagForTenant(@Nullable String tenantName) {
+    return getTagForTenant(OFFLINE_SERVER_TAG_SUFFIX, tenantName);
+  }
+
+  public static String getRealtimeTagForTenant(@Nullable String tenantName) {
+    return getTagForTenant(REALTIME_SERVER_TAG_SUFFIX, tenantName);
   }
 
-  public static String getRealtimeTagForTenant(String tenantName) {
+  private static String getTagForTenant(String tagSuffix, @Nullable String tenantName) {
     if (tenantName == null) {
-      return TagNameUtils.getRealtimeTagForTenant(DEFAULT_TENANT_NAME);
+      return DEFAULT_TENANT_NAME + tagSuffix;
     }
-    if (tenantName.endsWith(ServerType.REALTIME.toString())) {
+    if (tenantName.endsWith(tagSuffix)) {
       return tenantName;
     } else {
-      return TagNameUtils.buildRealtimeTagFromTenantName(tenantName);
+      return tenantName + tagSuffix;
     }
   }
 
-  public static String getOfflineTagForTenant(String tenantName) {
-    if (tenantName == null) {
-      return TagNameUtils.getOfflineTagForTenant(DEFAULT_TENANT_NAME);
-    }
-    if (tenantName.endsWith(ServerType.OFFLINE.toString())) {
-      return tenantName;
+  public static String extractBrokerTag(TenantConfig tenantConfig) {
```

Comment:
Can `tenantConfig` be null?

Reply:
It won't be (everything not annotated with nullable are non-null)